ui/packages/consul-ui/app/mixins/with-blocking-actions.js

```diff
@@ -107,5 +108,43 @@ export default Mixin.create({
         }
       );
     },
+    use: function(item) {
+      return this.repo
+        .findBySlug({
+          dc: this.modelFor('dc').dc.Name,
+          ns: get(item, 'Namespace'),
+          id: get(item, 'AccessorID'),
+        })
+        .then(item => {
+          return this.settings.persist({
+            token: {
+              AccessorID: get(item, 'AccessorID'),
+              SecretID: get(item, 'SecretID'),
+              Namespace: get(item, 'Namespace'),
+            },
+          });
+        });
+    },
+    logout: function(item) {
+      return this.settings.delete('token');
+    },
+    clone: function(item) {
+      let cloned;
+      return this.feedback.execute(() => {
+        return this.repo
+          .clone(item)
+          .then(item => {
+            cloned = item;
+            // cloning is similar to delete in that
+            // if you clone from the listing page, stay on the listing page
+            // whereas if you clone from another token, take me back to the listing page
+            // so I can see it
+            return this.afterDelete(...arguments);
+          })
+          .then(function() {
+            return cloned;
+          });
+      }, 'clone');
+    },
```

Comment:
I moved a few of our other mixins into here and deleted the old ones. All are mixins will be gone soon anyway, and they were originally only separate for organizations sake. Now we only have one `with-blocking-actions` Mixins left instead of a few.